### ISSUE TITLE
Some known issues

### ISSUE THREAD

**Issue by AhmerAfzal1** [beeline09/AdvancedPdfView#1]
I use this library, but 
- Search feature not working
- autoSpacing also not workig
- onError unable to catch protected file e.g. if (t is PdfPasswordException){}
- Display page not fitted in centered

Please take a look these issue

**Comment by beeline09** [comment]
This lib is still in development.

About search feature... If you want to highlight text on the page, use `pdfView.setSearchQuery("Google");`. If you want to parse text from the page, use `pdfView.parseText(arr, new PDFView.OnTextParseListener(){});` where `arr` is array of int page indexes for parse text

autoSpacing is a development of the author of pdfView and I did not develop it. I don't use it. I am quite satisfied with spacing. 

> onError unable to catch protected. 

What is happening at this moment? App crashes?

> Display page not fitted in centered

Please, attach screenshot.

**Comment by AhmerAfzal1** [comment]
App not crashes, but before I catch protected file using `onError` this code `if (t is PdfPasswordException){showDialogPassword()}`.
Suggestion to you that please use `CoroutineScope` in `DecodingRunner`

**Comment by beeline09** [comment]
Sorry for taking so long to reply, I have a lot of work to do...

Regarding the password, please send an example pdf with a password. I'll try to figure it out in my spare time.

I also updated the repository. Take a look at sample app. I made some changes and now everything should work. At the moment, in order to get page previews for your custom list, you need to add PdfView to your layout and call the desired callback.

For a full-fledged text search, you need to save all the text in the database using the `pdfView.parseText` method, then perform a search in the database. You can get the content using the `getTableOfContents` method. After that, you will have a good structure of the book in the database.

To highlight text, you must first assign a highlight color during pdfView initialization with the `setTextHighlightColor` method, after which you can call `pdfView.setSearchQuery("Bla-bla-bla");`

**Comment by beeline09** [comment]
> App not crashes, but before I catch protected file using `onError` this code `if (t is PdfPasswordException){showDialogPassword()}`. Suggestion to you that please use `CoroutineScope` in `DecodingRunner`

Sorry for taking so long to reply, I have a lot of work to do...

Regarding the password, please send an example pdf with a password. I'll try to figure it out in my spare time.

I also updated the repository. Take a look at sample app. I made some changes and now everything should work. At the moment, in order to get page previews for your custom list, you need to add PdfView to your layout and call the desired callback.

For a full-fledged text search, you need to save all the text in the database using the `pdfView.parseText` method, then perform a search in the database. You can get the content using the `getTableOfContents` method. After that, you will have a good structure of the book in the database.

To highlight text, you must first assign a highlight color during pdfView initialization with the `setTextHighlightColor` method, after which you can call `pdfView.setSearchQuery("Bla-bla-bla");`

**Comment by AhmerAfzal1** [comment]
I tested your code but `pdfView.setSearchQuery("quisque");` is not highlight proper, screenshot is attached. 
Please take a look this [library](https://github.com/gulshan98185/AndroidPdfViewer). Selection text and search text working, but not like charm.

![Screenshot 2022-02-18 153852](https://user-images.githubusercontent.com/11347099/154667042-52c707de-2a34-4a5b-8516-2d7b52c90ef8.png)

**Comment by beeline09** [comment]
> I tested your code but `pdfView.setSearchQuery("quisque");` is not highlight proper, screenshot is attached. Please take a look this [library](https://github.com/gulshan98185/AndroidPdfViewer). Selection text and search text working, but not like charm.
> 
> ![Screenshot 2022-02-18 153852](https://user-images.githubusercontent.com/11347099/154667042-52c707de-2a34-4a5b-8516-2d7b52c90ef8.png)

Yes, there are and will be problems, because it works like this:
pdfium gives us only the square of the first character of the line it found. I don't know why it's so idiotically done, but it's not up to me. It worked fine on other search queries, but something seems to be wrong with this one... I don't know the solution. If you find it, please make a request.

**Comment by AhmerAfzal1** [comment]
Last time I used this [library](https://github.com/gulshan98185/AndroidPdfViewer), Selection Text and Search Text worked fine, but it requires more concentration. I would be proud to help you add things to this great library. But I go abroad for a job. So I probably won't be able to spend much time with programming until I'm fully settled there. But I will check GitHub regularly to see the issues.

**Comment by meijieman** [comment]
https://github.com/gulshan98185/AndroidPdfViewer SOLVED MY PROBLEM. Ths
